homeassistant/components/velux/__init__.py

```diff
@@ -23,28 +24,45 @@
 )
 
 
-async def async_setup(hass, config):
-    """Set up the velux component."""
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Component setup, run import config flow for each entry in config."""
+    if DOMAIN not in config:
+        return True
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config[DOMAIN]
+        )
+    )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry):
+    """Set up Velux using config flow."""
     try:
-        hass.data[DATA_VELUX] = VeluxModule(hass, config[DOMAIN])
-        hass.data[DATA_VELUX].setup()
-        await hass.data[DATA_VELUX].async_start()
+        veluxModule = VeluxModule(hass, config_entry)
+        veluxModule.setup()
+        await veluxModule.async_start()
```

Comment:
`async_start()` may fail (see https://github.com/home-assistant/core/issues/34844), in which case you should raise ConfigEntryNotReady. That would really resolve #34844, I guess. The issue was closed due to inactivity.